pkg/stringhelper/stringhelper.go

```diff
@@ -36,6 +36,23 @@ func StringInSlice[T ~string](s []T, str T, contains bool) bool {
 	return false
 }
 
+// SubSlice checks if a slice exists within a slice
+func SubSlice[T ~string](s, sub []T) bool {
```

Comment:
I don't think this new implementation is doing the same as the old one, as it will only return true if the elements of the "sub" slice appear consecutive (and in the same order) in the slice "s". But this is not what the SCC test case is actually trying to check. See this UT:
```
		{ 
			testSliceA:     []string{"MKNOD", "NET_RAW", "SETUID", "SETGID", "KILL"},
			testSliceB:     []string{"MKNOD", "SETUID", "SETGID", "KILL"},
			expectedOutput: true,
		},
```
That unit test fails with your new version (SubSlice() returns false), but it should return true. All the elements in testSliceB appear in testSliceA, which is what we are actually checking in this SCC tc (if I'mt not wrong).
This other unit test should also make the SubsSlice() to return true:
```
		{
			testSliceA:     []string{"KILL", "SETUID", "MKNOD", "NET_RAW", "SETGID"},
			testSliceB:     []string{"MKNOD", "SETUID", "SETGID", "KILL"},
			expectedOutput: true,
		},
```

Reply:
Okay I re-did the SubSlice function to look at the individual elements and added another unit test case.